cachi2/core/package_managers/yarn/project.py

```diff
@@ -126,3 +142,46 @@ def yarn_cache(self) -> RootedPath:
     def from_source_dir(cls, source_dir: RootedPath) -> "Project":
         """Create a Project from a sources directory path."""
         return NotImplemented
+
+
+def get_semver_from_yarn_path(yarn_path: str) -> Optional[semver.version.Version]:
```

Comment:
Nitpick: if we're keeping these methods in the `project.py` file, maybe they can be methods in `YarnRc` and `PackageJson`? 

I feel they are a little more than just utility methods, since they have some yarn business logic copied into them, but since we're still doing the bulk of the processing in `main.py`, we're still keeping this module's ethos IMO.

Alternatively, they could be private methods in main.py.

Reply:
Well, since it seems no one has a strong preference either way, I'll leave my comment as suggestions, and go with whatever you feel it's best (including the current code).